src/xcode/ENA/ENA/Source/Scenes/Home/HomeTableViewController.swift

```diff
@@ -106,8 +106,16 @@ class HomeTableViewController: UITableViewController, NavigationBarOpacityDelega
 		viewModel.cclService.shouldShowNoticeTile
 			.receive(on: DispatchQueue.OCombine(.main))
 			.sink { [weak self] shouldShowNoticeTile in
+				let isHibernationState = CWAHibernationProvider.shared.isHibernationState
```

Comment:
Please pass the dependency in the constructor

```swift
init(cwaHibernationProvider: CWAHibernationProvider = CWAHibernationProvider.shared) { ... }
```

Reply:
Cause of a clean pattern style. In this way, you could also write a unit test and give a mocked version into it. Or a version with a mocked store into it. Thatswhy I write a second constructor for the singleton, that can take a mocked store. ;)

Reply:
As discussed, you can keep the singleton instance getter, without dependency injection as long you don't write a specific unit test with a mocked storage instance.